### DISCUSSION TITLE
Seeed Studio ESP32 S3 XIAO support ?

### DISCUSSION THREAD

**Discussion by kindmartin** [micropython/micropython#11899] [ESP32]
Hi all, I'm trying to make a FW that works for a XIAO S3 (https://www.seeedstudio.com/XIAO-ESP32S3-p-5627.html) , I've got almost all using that Octal PSRAM generic board as base board, but I to enable full psram but I cannot make sta_if works. 
it works for scan() but the XIAO don't connect to any working AP, it tested ok with several other ESP32 boards.

any hint ?

here some REPL 
ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x1 (POWERON),boot:0x8 (SPI_FAST_FLASH_BOOT)
SPIWP:0xee
mode:DIO, clock div:1
load:0x3fcd0108,len:0xf24
load:0x403b6000,len:0xc2c
load:0x403ba000,len:0x2d60
entry 0x403b61f4
MicroPython v1.20.0-111-g0000eb272-dirty on 2023-05-31; ESP32S3 module (spiram octal) with ESP32S3
Type "help()" for more information.
>>>
gc uos bdev
>>> import network
>>> sta_if = network.WLAN(network.STA_IF)
>>> sta_if.active(True)
True
>>> sta_if.scan()
[(b'FJS.Cocina.2Ghz', b'\xdc,n;C\xdd', 1, -54, 3, False), (b'iot', b'vM(Y]\xcf', 1, -54, 4, False), (b'FJS.Living.2Ghz', b'tM(Y]\xcf', 7, -69, 3, False), (b'DIRECT-PSLAPTOPTECODELLmsBN', b'\x96\xe7\x0b\xa5\x94\\', 7, -70, 3, False), (b'Fibertel WiFi689 2.4GHz', b'\xb0\xbb\xe5?1c', 11, -82, 3, False), (b'Fibertel WiFI224 2.4GHz', b'\x90XQx\x05p', 1, -83, 3, False), (b'Personal Wifi Zone', b'\x92XQx\x05t', 1, -83, 0, False), (b'Pepe.Cocina.2Ghz', b'\x80*\xa8\x17\x1cS', 6, -84, 4, False)]
>>> sta_if.scan()
[(b'MartinAP', b'\x98\xde\xd0\x156\xd2', 11, -47, 3, False), (b'iot', b'vM(Y]\xcf', 1, -59, 4, False), (b'FJS.Cocina.2Ghz', b'\xdc,n;C\xdd', 1, -60, 3, False), (b'DIRECT-PSLAPTOPTECODELLmsBN', b'\x96\xe7\x0b\xa5\x94\\', 7, -68, 3, False), (b'FJS.Living.2Ghz', b'tM(Y]\xcf', 7, -68, 3, False), (b'Mkt.Altillo.root', b'\xb8i\xf4(\xdc?', 8, -79, 3, False), (b'Mkt.PlantaAlta.2Ghz', b'\xbai\xf4(\xdc?', 8, -79, 3, False), (b'FJS.IoT', b'\xbai\xf4(\xdc@', 8, -79, 3, False), (b'Personal Wifi Zone', b'\x92XQx\x05t', 1, -82, 0, False), (b'Fibertel WiFi689 2.4GHz', b'\xb0\xbb\xe5?1c', 11, -83, 3, False), (b'Fibertel WiFI224 2.4GHz', b'\x90XQx\x05p', 1, -85, 3, False), (b'Fjs.modem.claro.2Ghz', b'(ww\x8b\x93\x82', 5, -91, 4, False), (b'Fibertel WiFi689 2.4GHz_EXT', b'`2\xb1;\xfc\xa8', 11, -91, 4, False)]
>>> sta_if.connect('MartinAP','88888888')
>>> sta_if.ifconfig()
('0.0.0.0', '0.0.0.0', '0.0.0.0', '0.0.0.0')
>>> sta_if.ifconfig()
('0.0.0.0', '0.0.0.0', '0.0.0.0', '0.0.0.0')
>>> machine.unique_id()
b'4\x85\x18\xac\xbd0'

**Comment by shariltumin** [comment]
These are my PSRAM config def:

```
# SPI RAM config
#
CONFIG_ESP32S2_SPIRAM_SUPPORT=y
CONFIG_ESP32S3_SPIRAM_SUPPORT=y
CONFIG_SPIRAM_MODE_OCT=y
CONFIG_SPIRAM_TYPE_AUTO=y
CONFIG_DEFAULT_PSRAM_CLK_IO=30
CONFIG_DEFAULT_PSRAM_CS_IO=26
CONFIG_SPIRAM_SPEED_80M=y
CONFIG_SPIRAM=y
CONFIG_SPIRAM_BOOT_INIT=y
CONFIG_SPIRAM_IGNORE_NOTFOUND=y
CONFIG_SPIRAM_SIZE=-1
CONFIG_SPIRAM_USE_MALLOC=y
CONFIG_SPIRAM_MEMTEST=y
CONFIG_SPIRAM_CACHE_WORKAROUND=y
```

It seem to work just fine:

```
>>> 
>>> import streamin_server
Camera ready?:  True
Waiting ...
Waiting ...
Connected to dlink-3530
network config: ('192.168.4.70', '255.255.252.0', '192.168.4.1', '192.168.4.1')
Request from: ('192.168.4.27', 46250)
TCP send error [Errno 104] ECONNRESET
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "streamin_server.py", line 90, in <module>
KeyboardInterrupt: 
>>> 
MPY: soft reboot
MicroPython v1.20.0-242-g6a9db521e on 2023-06-29; ESP32S3-XIAO OV2640 (KAKI5) with ESP32-S3
>>> import machine
>>> machine.reset()
```

The firmware was compiled with IDF-510. However, I could not get the esp console log to show up in the REPL.

The best placement for a Wi-Fi antenna is vertical with the short edge pointing up.

**Reply by virgileMaon** [reply]
I'm also insterested with the support of micropython for the xiao esp32-s3

**Comment by kindmartin** [comment]
thanks, I was trying an already compiled fw from [micropython.org/download/](https://micropython.org/download/GENERIC_S3_SPIRAM_OCT) and works so my tool chain seems not to works.  I will investigate

**Comment by steveh127** [comment]
I use the generic ESP32 S3 build, not the Octal spiram version, with these XIAO ESP32 S3 boards and have no problems with network functions, or anything else for that matter. It may not utilise all available memory but I've seen no problems with it. 

As an aside, am I the only person who has no idea what 'FW' stands for?

**Reply by davefes** [reply]
HW/SW/FW

**Comment by sosi-deadeye** [comment]
Steps to compile:

```
# ENV
# esp-idf: ~/esp-idf
# micropython: ~/micropython


# ESP-IDF v5.0.4
source esp-idf/export.sh

cd micropython

# removing managed_components. Somethimes it does not work if you
# reuse the same env to compile different boards
# then it seems, that managed_components contains files, which are not
# required
rm -R ports/esp32/managed_components

git pull
make -C mpy-cross -j 16
make -C ports/esp32 BOARD=ESP32_GENERIC_S3 BOARD_VARIANT=SPIRAM_OCT clean
make -C ports/esp32 BOARD=ESP32_GENERIC_S3 BOARD_VARIANT=SPIRAM_OCT submodules
make -C ports/esp32 BOARD=ESP32_GENERIC_S3 BOARD_VARIANT=SPIRAM_OCT -j 16

```

You can also look into: https://github.com/micropython/micropython/blob/master/tools/ci.sh
